group/views.py

```diff
@@ -1,22 +1,157 @@
+            if request.user != group.creator:
+                raise PermissionDenied("Access Denied")
+            return group
+        except Group.DoesNotExist as exc:
+            raise Http404 from exc
 
+    def get(self, request):
+        """Retrieve a list of all the groups"""
+        groups = Group.objects.all()
+        serializer = GroupSerializer(groups, many=True, context={'request': request}) # added request context
+        return Response(serializer.data, status=status.HTTP_200_OK)
 
-@api_view(['POST'])
-@permission_classes([IsAuthenticated])
-def create_group(request):
-    """function based view for creating a group"""
-    if request.method == 'POST':
-        data = request.data.copy()
-        data['creator'] = request.user.id
+    def post(self, request):
+        """Create a group"""
+        data = request.data
+
+        # user_id = request.user.id
+        user_serializer = UserSerializer(request.user)
+        user_data = user_serializer.data
+        user = request.user
+        print(f"{user_data} ldldldl")
+        data['creator'] = user.id
+        data['members'] = [user_data]
         serializer = GroupSerializer(data=data)
         if serializer.is_valid():
             serializer.save()
             return Response(serializer.data, status=status.HTTP_201_CREATED)
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
-    return None
+
+    # def post(self, request):
+    #     """Create a group"""
+    #     data = request.data
+    #     user_id = request.user.id
+    #     data['creator'] = user_id
+    #     data['members'] = [user_id]
+    #     serializer = GroupSerializer(data=data)
+    #     if serializer.is_valid():
+    #         serializer.save()
+    #         return Response(serializer.data, status=status.HTTP_201_CREATED)
+    #     return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    def patch(self, request):
+        """Update members list"""
+        data = request.data
+        group_id = data.get("group_id")
+
+        data.pop('creator', None)
+        data.pop('name', None)
+        try:
+            group = self.get_object(request, group_id)
+            errors = []
+            for member in data.get("members"):
+                try:
+                    user = User.objects.get(id=member)
+                    if user.id not in group.members.all()\
+                            .values_list('id', flat=True):
+                        group.members.add(user)
+                except User.DoesNotExist:
+                    errors.append(f"User with ID {member} is not found")
+            serializer = GroupSerializer(group, data=data, partial=True)
+            if serializer.is_valid():
+                serializer.save()
+                return Response(serializer.data, status=status.HTTP_200_OK)
+            if errors:
+                return Response({"error": errors},
+                                status=status.HTTP_404_NOT_FOUND)
```

Comment:
Ppatch should not be returning 404... `If the PATCH method was unsuccessful, status codes such as 304 (Not Modified), 400 (Bad Request), or 422 (Unprocessable Entity) may be seen.`